packages/client/hmi-client/src/components/responsive-matrix/matrix.vue

```diff
@@ -0,0 +1,902 @@
+			uGridColDisplayTransition: 12,
+
+			// row/col data
+			uNumRow: this.numRows,
+			uNumCol: this.numCols,
+			uMicroRow: this.uniforms.uMicroRow,
+			uMicroCol: this.uniforms.uMicroCol,
+
+			// cell element color data
+			uColor: Texture.fromBuffer(color, this.numCols, this.numRows),
+
+			uTime: 0
+		};
+
+		// set callback to update uniforms every render call
+		app.ticker.add(this.handleTick);
+
+		// run shader on quad
+		const quadShader = Shader.from(matrixVS, matrixFS, this.uniforms);
+		const quad = new Mesh(geometry, quadShader);
+
+		// center quad in world
+		quad.position.set(
+			(this as any).$viewport.worldWidth / 2,
+			(this as any).$viewport.worldHeight / 2
+		);
+
+		// add quad to viewport
+		(this as any).$viewport.addChild(quad as any);
+
+		// run shader on grid
+		// const gridShader = Shader.from(matrixVS, matrixGridFS, this.uniforms);
+		// const grid = new Mesh(geometry, gridShader);
+
+		// center grid in world
+		// grid.position.set(
+		// 	(this as any).$viewport.worldWidth / 2,
+		// 	(this as any).$viewport.worldHeight / 2
+		// );
+
+		// add grid to viewport
+		// (this as any).$viewport.addChild(grid as any);
+
+		// center and zoom camera to world
+		this.centerGraph();
+
+		// add pixi interaction handlers
+		(this as any).$viewport.on('pointerdown', this.handleMouseDown);
+		(this as any).$viewport.on('pointerup', this.handleMouseUp);
+		(this as any).$viewport.on('moved' as any, this.incrementMove);
+		(this as any).$viewport.on('zoomed-end' as any, this.incrementUpdate);
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// watch                                                                        //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	watch: {
+		data() {
+			this.processData();
+			this.processLabels();
+			this.processActiveCells();
+		},
+		cellLabelRow() {
+			this.processLabels();
+		},
+		cellLabelCol() {
+			this.processLabels();
+		}
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// unmounted                                                                    //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	unmounted() {
+		this.resizeObserver.disconnect();
+		window.removeEventListener('keydown', this.handleKey);
+		window.removeEventListener('keyup', this.handleKey);
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// methods                                                                      //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	methods: {
```

Comment:
All those methods could be put in a `services/responsive-matrix.ts` file

Reply:
Let's do these as a separate round after things have settled in a bit and has demonstrated the utility